magento/pwa-studio
# Service Worker HTML Handling

## Description

Modified the HTML handling capabilities of the Service Worker to fetch and return the same HTML file irrespective of the page the client is in. This is because HTML for `/` is the same `venia-bottoms.html` or `athena-tank-dress.html`. Also since the same HTML is served for all the routes, offline capabilities are improved compared to `develop` as demonstrated in the demo gif.

## Related Issue
Closes #1781 

### Verification Steps
1. Start the server using `yarn build && yarn stage:venia`.
2. Open the app once the server is ready.
3. Navigate between multiple pages like category, product, etc. The server should only get requests for `/` unline `develop` in which each page will request its own HTML file.

## Screenshots / Screen Captures (if appropriate)
![html caching demo-low-res](https://user-images.githubusercontent.com/35203638/66517801-69fbe800-eaa9-11e9-8f7c-1d36d8be08e1.gif)

## Checklist
* `venia:stage and venia:watch` should both work as `develop`. No business logic changed.
* Merge only after #1832 and #1859 are merged.
